src/react-native-cool-modals/createNativeStackNavigator.js

```diff
@@ -1,15 +1,33 @@
 import {
   createNavigatorFactory,
+  StackRouter as OldStackRouter,
   StackActions,
-  StackRouter,
   useNavigationBuilder,
 } from '@react-navigation/native';
 import * as React from 'react';
 
 import NativeStackView from './NativeStackView';
+import logger from 'logger';
 
 function NativeStackNavigator(props) {
   const { children, initialRouteName, screenOptions, ...rest } = props;
+  const StackRouter = (...args) => {
+    const oldRouter = OldStackRouter(...args);
+    return {
+      ...oldRouter,
+      getStateForAction(state, action, options) {
+        if (action.type === 'PUSH') {
+          if (
+            state.routes[state.routes.length - 1].name === action.payload.name
```

Comment:
Just a heads up:  When we open the DPI sheet  (or an LP Token expanded state) and then you tap on one underlying asset we're technically pushing the same route twice, so not sure if this would be breaking that behavior or not.